src/mlpack/methods/linear_regression/linear_regression.hpp

```diff
@@ -87,24 +110,211 @@ class LinearRegression
    * To set the regularization parameter lambda, call Lambda() or set a
    * different value in the constructor.
    *
+   * This version of `Train()` is deprecated and will be removed in mlpack
+   * 5.0.0.  Use the version of `Train()` that specifies `lambda` before
+   * `intercept` instead.
+   *
    * @param predictors X, the matrix of data points to train the model on.
    * @param responses y, the responses to the data points.
    * @param weights Observation weights (for boosting).
    * @param intercept Whether or not to fit an intercept term.
    * @return The least squares error after training.
    */
+  mlpack_deprecated /** Will be removed in mlpack 5.0.0. */
   double Train(const arma::mat& predictors,
                const arma::rowvec& responses,
                const arma::rowvec& weights,
-               const bool intercept = true);
+               const bool intercept);
+
+  /**
+   * Train the LinearRegression model.  This is a dummy overload so that
+   * MetaInfoExtractor can properly detect that LinearRegression is a regression
+   * method.
+   */
+  template<typename MatType>
+  ElemType Train(const MatType& predictors,
+                 const arma::rowvec& responses);
+
+  /**
+   * Train the LinearRegression model on the given data and weights. Careful!
+   * This will completely ignore and overwrite the existing model. This
+   * particular implementation does not have an incremental training algorithm.
+   * To set the regularization parameter lambda, call Lambda() or set a
```

Comment:
Finally, I got confused a bit with the terminology I was thinking about the `weights` matrix as the `parameters()`. Then I had to go over the code. 

I do not know how to prevent this confusion, but it got me for a while.

Reply:
I'm not sure I understand the confusion... can you clarify what you were confused about?  I can definitely try and address the documentation.

For what it's worth, some mlpack methods do support incremental training (so, a second call to `Train()` starts from the existing model, instead of starting over), so I've been careful in the documentation to note which is the case for each method.

Reply:
Are weights related to Weighted Linear regression?

Reply:
Ah, I see; yeah, in this case, the weights correspond only to instance weights, i.e., weights for each individual point.  I updated the documentation in c194fb73f8.  It's not quite the same thing as weighted linear regression (I think).  Hopefully it's a bit clearer now, let me know if I can improve it further.